### PR TITLE
[ADD] base, base_setup, web, web_editor, web_unsplash

### PR DESCRIPTION
**Pull request by kos94ok-3D**
My first steps in the migration

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by kos94ok-3D** [comment]
Hi, @MiquelRForgeFlow.
Can you help me? Is it a problem with Travis or I missing something?

**Comment by MiquelRForgeFlow** [comment]
I'm busy with other things, sorry but it's not my priority right now

**Comment by kos94ok-3D** [comment]
Hi, @StefanRijnhart .
Have you any time in order to help me? Is it a problem with Travis or I missing something?

**Comment by kos94ok-3D** [comment]
HI, @pedrobaeza.
Maybe you have any idea about an error in Travis.

**Comment by pedrobaeza** [comment]
Sorry, I'm not either on this.

**Comment by StefanRijnhart** [comment]
Very nice work, thank you! Travis failure might be related to an issue with the port of the OpenUpgrade framework to Odoo 13, c.q. running Openupgrade specific tests (of which there are none yet). Maybe you can reproduce the test error by running locally on an odoo 12 demodatabase with the OPENUPGRADE_TESTS=1 environment variable set?

**Comment by kos94ok-3D** [comment]
> Very nice work, thank you! Travis failure might be related to an issue with the port of the OpenUpgrade framework to Odoo 13, c.q. running Openupgrade specific tests (of which there are none yet). Maybe you can reproduce the test error by running locally on an odoo 12 demodatabase with the OPENUPGRADE_TESTS=1 environment variable set?

Thank you.
I found two problems:

1. importlib.import_module can not correctly work with names which contains dots
2. tools.config['test_tags'] return None if not set so we get Traceback

What to do? Someone use migration tests? (I fixed it in #2207)

**Comment by kos94ok-3D** [comment]
> Very nice work, thank you! Travis failure might be related to an issue with the port of the OpenUpgrade framework to Odoo 13, c.q. running Openupgrade specific tests (of which there are none yet). Maybe you can reproduce the test error by running locally on an odoo 12 demodatabase with the OPENUPGRADE_TESTS=1 environment variable set?

Thank you.
I found two problems:

1. importlib.import_module can not correctly work with names which contains dots
2. tools.config['test_tags'] return None if not set so we get Traceback

What to do? Someone use migration tests?

**Comment by kos94ok-3D** [comment]
@StefanRijnhart 
I fix it in #2207

File: odoo/addons/base/migrations/13.0.1.3/openupgrade_analysis_work.txt
Comment (StefanRijnhart) [inline odoo/addons/base/migrations/13.0.1.3/openupgrade_analysis_work.txt:111]: This implies a column conversion from INTEGER to VARCHAR. Does fields.Selection support this conversion?

Comment (kos94ok-3D) [reply odoo/addons/base/migrations/13.0.1.3/openupgrade_analysis_work.txt:111]: I fixed it.

File: odoo/addons/base/migrations/13.0.1.3/pre-migration.py
Comment (StefanRijnhart) [inline odoo/addons/base/migrations/13.0.1.3/pre-migration.py]: Please rename to `None`. It will be renamed to the result of `openupgrade.get_legacy_name('week_start')` (and you can use that method to retrieve the renamed column name when you process its values in the post script).

That way, it is easy to determine what the origin of the field is when it is encountered in the database scheme.

Comment (kos94ok-3D) [reply odoo/addons/base/migrations/13.0.1.3/pre-migration.py]: Done

**Review by StefanRijnhart — APPROVED** [review]
Great work, thanks! I had been doing a bit of work on this module and I have a small number of minor additions that I will propose when this is merged.

**Comment by kos94ok-3D** [comment]
> Great work, thanks! I had been doing a bit of work on this module and I have a small number of minor additions that I will propose when this is merged.

Thank you for review.
OK

**Comment by MiquelRForgeFlow** [comment]
@StefanRijnhart Is there such a hurry as to not wait for other opinions before merge?

One thing is that I said I can't "help" because I'm busy with other things and another thing is that I can't say something before it gets merged. It's not a trivial PR and more work needs to be done in `base`.

**Comment by pedrobaeza** [comment]
Yeah, I agree, because in documentation it mentions now that `base` module is Done, and possible issues that arises derived from it must be now attended. Better to remove that tag or put something like "tentative".

**Comment by StefanRijnhart** [comment]
Well, again I think projects like openupgrade benefit from optimistic merging, in which incomplete and suboptimal code can be merged if it constitutes an improvement and when tests are green.

**Comment by pedrobaeza** [comment]
As said, the problem is to qualify this "experimental" code (although of course it's a great achievement from a new contributor) as "Done" in the documentation.

**Comment by StefanRijnhart** [comment]
Follow up here: https://github.com/OCA/OpenUpgrade/pull/2210

**Pull request merged**
